File: pdu/python/drone_srv_msgs/pdu_cdr_conv_DroneLandRequestPacket.py
from .pdu_pytype_DroneLandRequestPacket import DroneLandRequestPacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..drone_srv_msgs.pdu_cdr_conv_DroneLandRequest import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceRequestHeader import *



def py_to_cdr_body_DroneLandRequestPacket(writer: CdrWriter, src: DroneLandRequestPacket):
    py_to_cdr_body_ServiceRequestHeader(writer, src.header)
    py_to_cdr_body_DroneLandRequest(writer, src.body)


def cdr_body_to_py_DroneLandRequestPacket(reader: CdrReader, dst: DroneLandRequestPacket):
    cdr_body_to_py_ServiceRequestHeader(reader, dst.header)
    cdr_body_to_py_DroneLandRequest(reader, dst.body)
    return dst


def py_to_cdr_DroneLandRequestPacket(src: DroneLandRequestPacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DroneLandRequestPacket(writer, src)
    return writer.bytes()


def cdr_to_py_DroneLandRequestPacket(cdr_payload) -> DroneLandRequestPacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DroneLandRequestPacket()
    return cdr_body_to_py_DroneLandRequestPacket(reader, dst)

File: pdu/javascript/drone_srv_msgs/pdu_cdr_conv_DroneLandRequestPacket.js
import { DroneLandRequestPacket } from './pdu_jstype_DroneLandRequestPacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { DroneLandRequest } from '../drone_srv_msgs/pdu_jstype_DroneLandRequest.js';
import { PduDroneLandRequestConverter } from '../drone_srv_msgs/pdu_cdr_conv_DroneLandRequest.js';
import { ServiceRequestHeader } from '../hako_srv_msgs/pdu_jstype_ServiceRequestHeader.js';
import { PduServiceRequestHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceRequestHeader.js';


export class PduDroneLandRequestPacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DroneLandRequestPacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceRequestHeaderConverter.to_cdr_body(writer, src.header);
        PduDroneLandRequestConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DroneLandRequestPacket } dst
     * @returns { DroneLandRequestPacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceRequestHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduDroneLandRequestConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { DroneLandRequestPacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DroneLandRequestPacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DroneLandRequestPacket());
    }
}
